### PR TITLE
Add digilines to teleport tube and refactor code

### LINKED ISSUE #46: repairing teleport tubes w/ a hammer doesn't restore them to functionality.
a player reports:

When a sender teleport tube (tp) is broken, use hammer to repair. Now the tp tube sender is not working anymore. Expected: tp tube sender should tp the items as it did before breaking, doesn't work even after setting the channel name again. Workaround: destroy the receiver tube and then place it again and configure to get the sender tp tube working again.

### LINKED ISSUE #47: Cleanup tp tubes a bit
Remove:
https://github.com/mt-mods/pipeworks/blob/8c251800db1d9d190829cfdeea8cf339bcdecc89/teleport_tube.lua#L124-L136

Also remove these everywhere and just use direct `tp_tube_db`, just load db during mod loading stage instead. No real reason for deferred db read:
https://github.com/mt-mods/pipeworks/blob/8c251800db1d9d190829cfdeea8cf339bcdecc89/teleport_tube.lua#L114

Tp tube table `tp_tube_db` is local to file, recreating local for function scope in most cases also isn't helping with performance but instead just add unnecessary LoC.

### PR DESCRIPTION
**Pull request by OgelGames**
Adds the capability to set the teleport tube channel via digiline message, respecting private channels by adding an `owner` field on placement. Also refactors the code significantly to improve readability and maintainability.

Fixes #46 and closes #47

@BuckarooBanzay @S-S-X @TurkeyMcMac

### CODE REVIEW

File: teleport_tube.lua [hunk 1]
```diff
@@ -1,206 +1,304 @@
...
-	return {x = pos.x, y = pos.y, z = pos.z, cr = cr, channel = channel}
+local function deserialize_tube(hash, str)
+	local sep = str:sub(2, 2) == ":"
+	local cr = tonumber(str:sub(1, 1))
+	local channel = str:sub(3)
+	if sep and cr and channel then
+		local pos = minetest.get_position_from_hash(tonumber(hash))
+		return {x = pos.x, y = pos.y, z = pos.z, cr = cr, channel = channel}
+	end
 end
```
Comment (fluxionary) [inline teleport_tube.lua:33]: i'm not intimate w/ this code, but these don't seem equivalent at all. is "cr" *always* a single digit? why not do something like
```lua
local cr, channel = unpack(str:split(":", true, 1))
cr = tonumber(cr)
if cr and channel then
    local pos = minetest.get_position_from_hash(tonumber(hash))
    pos.cr = cr
    pos.channel = channel
    return pos
else
    return minetest.deserialize(str)
end
```

Comment (fluxionary) [reply teleport_tube.lua:33]: oh i see it's "can receive" and is either 0 or 1? so i guess it's always a single number.

Comment (fluxionary) [reply teleport_tube.lua:33]: and the de-serialize stuff is handled in the update code. got it.

Comment (OgelGames) [reply teleport_tube.lua:33]: Yes, `cr` (short for `can_receive`) is always either 0 or 1, so it's much more efficient to use `string.sub`.

**Review by wsor4035 — APPROVED** [review]
did not test it(can if you wish), but reading over the code, it makes sense

**Review by BuckarooBanzay — APPROVED** [review]

**Pull request merged**